src/test/catch-main.cpp

```diff
@@ -0,0 +1,4 @@
+#ifndef CATCH_CONFIG_MAIN
```

Comment:
1. Can we move this file to Utilities? I just feel creating one folder for only one file may not be necessary.
2. It is a cpp file instead of a header file, do we still need #ifndef #endif protection?

Reply:
Yes the ifndef is a weird catch thing.  It needs to see that definition just once in one source file for a multi file test program. https://github.com/catchorg/Catch2/blob/master/docs/tutorial.md#scaling-up
In my code I intend other files in test for source in src.

Reply:
catch-main.cpp is always responsible to handle the main() and none of the test_XX.cpp should add a main. catch-main.cpp should never been included by others. In this scenario, I don't think we need the protection.

Reply:
https://github.com/catchorg/Catch2/blob/master/docs/slow-compiles.md#practical-example
test-main.cpp doesn't need the protection.

Reply:
I see your point I'll fix it.